src/BaseListView.php

```diff
@@ -342,11 +343,10 @@ private function prepareDataReaderAndGetItems(array $filters): array
             $this->urlConfig->getSortParameterType(),
         );
 
-        $this->preparedDataReader = $this->prepareDataReaderByParams($page, $previousPage, $pageSize, $sort, $filters);
-
         try {
+            $this->preparedDataReader = $this->prepareDataReaderByParams($page, $previousPage, $pageSize, $sort, $filters);
             return $this->getItems($this->preparedDataReader);
-        } catch (PageNotFoundException $exception) {
+        } catch (PageNotFoundException|PaginatorException $exception) {
```

Comment:
`pageNotFoundExceptionCallback` work with `PageNotFoundException` and don't support `PaginatorException`. It needs to resolve.